### DISCUSSION TITLE
Order Details item not save without click on Update button

### DISCUSSION THREAD

**Discussion by ghost** [serenity-is/Serenity#6037] [Help] [answered]
After adding items to the New Order Details, the items were not saved to the database without click the update button.   Can we make it auto save or another other option?  Some user may just exist and items get lost.
![image](https://user-images.githubusercontent.com/44040606/137522528-6d457c51-e790-4e1a-89b4-486286520f05.png)
 the browser without click the update button.

**Comment by VictorTomaili** [comment]
order detail has a callback, you can save this list manually on this callback. But let the handler for handle it on save by default. You can also check for the form is dirty before closing the dialog to prevent closing dialog when some details have changed.

**Reply by ghost** [reply]
Thank you for response.  How to check the form is dirty before closing the dialog.  Do you have the sample codes?

**Reply by go2ismail** [reply]
Hello @elemechinc 
Maybe you can add pendingChangesConfirmation
` constructor() {
            super();

            this.ordersGrid = new CustomerOrdersGrid(this.byId('OrdersGrid'));
            // force order dialog to open in Dialog mode instead of Panel mode
            // which is set as default on OrderDialog with @panelAttribute
            this.ordersGrid.openDialogsAsPanel = false; 

            this.byId('NoteList').closest('.field').hide().end().appendTo(this.byId('TabNotes'));
            Serenity.Extensions.DialogUtils.pendingChangesConfirmation(this.element, () => this.getSaveState() != this.loadedState);
        }`

detail can be found at northwind example: https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Customer/CustomerDialog.ts

**Comment by ghost** [comment, marked as answer]
Thanks! It is working prefect.
